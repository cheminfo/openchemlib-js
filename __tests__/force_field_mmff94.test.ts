import { beforeAll, describe, expect, it } from 'vitest';

import {
  ConformerGenerator,
  ForceFieldMMFF94,
  Molecule,
  Resources,
} from '../lib/index.js';

beforeAll(() => {
  Resources.registerResourcesNodejs();
});

describe('ForceFieldMMFF94', () => {
  it('should generate force field', () => {
    const mol = Molecule.fromSmiles('COCCON');
    const gen = new ConformerGenerator(1);
    gen.getOneConformerAsMolecule(mol);
    const molfileBefore = mol.toMolfile();

    const ff = new ForceFieldMMFF94(mol, 'MMFF94');
    ff.minimise();
    const molfileAfter = mol.toMolfile();
    expect(molfileAfter).toMatchSnapshot();
    expect(molfileAfter).not.toBe(molfileBefore);
  });
});
