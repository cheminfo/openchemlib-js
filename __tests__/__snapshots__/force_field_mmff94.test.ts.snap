// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForceFieldMMFF94 > should generate force field 1`] = `
"#1
Actelion Java MolfileCreator 1.0

 15 14  0  0  0  0  0  0  0  0999 V2000
    1.7685    0.9053   -1.5976 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.4013   -0.3647   -1.5301 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.6128   -1.3588   -0.8641 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.6782   -1.2484    0.6601 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.0034   -1.5294    1.1322 O   0  0  0  0  0  0  0  0  0  0  0  0
    3.1221   -2.9596    1.4011 N   0  0  0  0  0  0  0  0  0  0  0  0
    0.8075    0.8298   -2.1152 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.4153    1.5787   -2.1667 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.6276    1.3287   -0.5999 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.0142   -2.3277   -1.1788 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.5751   -1.3011   -1.2119 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.9515   -1.9235    1.1286 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.4393   -0.2333    0.9888 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.4846   -2.9468    2.3562 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.9205   -3.1978    0.8087 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  1  7  1  0  0  0  0
  1  8  1  0  0  0  0
  1  9  1  0  0  0  0
  3 10  1  0  0  0  0
  3 11  1  0  0  0  0
  4 12  1  0  0  0  0
  4 13  1  0  0  0  0
  6 14  1  0  0  0  0
  6 15  1  0  0  0  0
M  END
"
`;
